examples/capture/main.rs

```diff
@@ -0,0 +1,98 @@
+/// This example shows how to capture an image by rendering it to a texture, copying the texture to
+/// a buffer, and retrieving it from the buffer. This could be used for "taking a screenshot," with
+/// the added benefit that this method doesn't require a window to be created.
+fn main() {
+    env_logger::init();
+
+    let instance = wgpu::Instance::new();
+
+    let adapter = instance.get_adapter(&wgpu::AdapterDescriptor {
+        power_preference: wgpu::PowerPreference::LowPower,
+    });
+
+    let mut device = adapter.request_device(&wgpu::DeviceDescriptor {
+        extensions: wgpu::Extensions {
+            anisotropic_filtering: false,
+        },
+        limits: wgpu::Limits::default(),
+    });
+
+    // Rendered image is 256×256 with 32-bit BGRA color
+    let size = 256u32;
+
+    // The output buffer lets us retrieve the data as an array
+    let output_buffer = device.create_buffer(&wgpu::BufferDescriptor {
+        size: (size * size * std::mem::size_of::<u32>() as u32) as u64,
+        usage: wgpu::BufferUsage::MAP_READ,
```

Comment:
I think we also need `BufferUsage::TRANSFER_DEST` as we are performing a `copy_texture_to_buffer` on this buffer.

Reply:
Done